addons/stock/models/stock_location.py

```diff
@@ -466,3 +466,13 @@ def toggle_active(self):
         for route in self:
             route.with_context(active_test=False).rule_ids.filtered(lambda ru: ru.active == route.active).toggle_active()
         super().toggle_active()
+
+    @api.constrains('company_id', 'rule_ids')
+    def _check_rule_ids(self):
+        for record in self:
+            if not record.company_id:
+                continue
+
+            for rule in record.rule_ids:
+                if record.company_id.id != rule.company_id.id:
```

Comment:
```suggestion
            if any(rule.company_id != route.company_id for rule in route.rule_ids)
```

Reply:
Yes it's indeed cleaner. but we can't substitute `rule` in the text placeholder if we do so